test/units/TEST-50-DISSECT.sysext.sh

```diff
@@ -114,7 +114,11 @@ prepare_extension_image() {
     mkdir -p "$ext_dir/$hierarchy"
     touch "$ext_dir$hierarchy/preexisting-file-in-extension-image"
 
-    prepend_trap "rm -rf ${ext_dir@Q}"
+    # For issue #32965
+    touch "$root/var/lib/extensions/empty-image.raw"
+    mkdir "$root/var/lib/extensions/empty-directory"
+
+    prepend_trap "rm -rf ${ext_dir@Q} $root/var/lib/extensions/empty-image.raw $root/var/lib/extensions/empty-directory"
```

Comment:
Instead of I would extending this helper maybe an explicit test would be better.
For empty directories there's a test already to cover their masking behavior (that's documented in the man page, but empty image files not) in `Test that an extension consisting of an empty directory under /etc/extensions/ takes precedence`.